---
title: "2.correlation_btween_readsandflowC"
author: "Deepan"
date: "12/18/2024"
output: html_document
---

## Load libraries

```{r, message = FALSE}
# Packages I commonly use
# install.packages("stringr")
# install.packages("ggplot2")
# install.packages("reshape")
# install.packages("reshape2")
# install.packages("gplots")
# install.packages("FactoMineR")
# install.packages("factoextra")
library(stringr)
library(ggplot2)
library(FactoMineR)
library(factoextra)
library(reshape)
library(reshape2)
library(gplots)
library(ggpubr)
library(RColorBrewer)
library(dplyr)
```

## Define color vectors

```{r}

bpart_colors <- c("darkgoldenrod", "lightskyblue3", "indianred")

indiv_colors_RPCA <- c("1" = "firebrick4", "2" = "darkolivegreen", "3" = "midnightblue", "4" = "coral1", "5" = "lightpink1", "6" = "lightskyblue3", "7" = "goldenrod1", "8" = "sienna3", "9" = "olivedrab4", "10" = "slateblue2", "11" = "burlywood4")

indiv_colors <- c("firebrick4", "darkolivegreen", "midnightblue", "coral1", "lightpink1", "lightskyblue3",  "goldenrod1", "sienna3", "olivedrab4", "slateblue2", "burlywood4")

bpart_colors_RPCA <- c("Sebaceous" = "darkgoldenrod", "Moist" = "lightskyblue3", "Dry" = "indianred")

vec_col0 <- c("indianred"  ,  "gold"      ,    "rosybrown4"  ,   "powderblue"      ,     "lightsteelblue2"     ,     "lightpink"   ,   "darkseagreen4" , "palegreen"  ,    "peachpuff3" , "khaki4"    ,     "orchid4" , "darkolivegreen", "rosybrown3"  ,   "palevioletred2" ,
              "darkorange1"    ,      "paleturquoise3" , "khaki"     ,     "dodgerblue4" ,   "sienna2"    ,    "darkgreen"   ,   "lightsalmon"  ,  "orchid2" ,   "lightsteelblue4"  ,   "lightsalmon2" ,  "lemonchiffon3"  , "honeydew"  ,   "sienna4"   ,  "darkolivegreen2"    ,    "gold1", "ivory3")

# sample order vector for paired analysis

sample_order <- c("1-AC-PMA", "1-AC", "1-FH-PMA", "1-FH", "1-UB-PMA", "1-UB", "2-AC-PMA", "2-AC", "2-FH-PMA", "2-FH", "2-UB-PMA", "2-UB", "3-FH-PMA", "3-FH", "3-PC-PMA", "3-PC", "3-UB-PMA", "3-UB", "4-AC-PMA", "4-AC", "4-FA-PMA", "4-FA", "4-FH-PMA", "4-FH", "4-PC-PMA", "4-PC", "4-UB-PMA", "4-UB", "5-AC-PMA", "5-AC", "5-FA-PMA", "5-FA", "5-PC-PMA", "5-PC", "5-UB-PMA", "5-UB", "6-AC-PMA", "6-AC", "6-FH-PMA", "6-FH", "6-PC-PMA", "6-PC", "6-UB-PMA", "6-UB", "7-FA-PMA", "7-FA", "7-FH-PMA", "7-FH", "7-PC-PMA", "7-PC", "7-UB-PMA", "7-UB", "8-AC-PMA", "8-AC", "8-Abd-PMA",  "8-Abd",  "8-FA-PMA", "8-FA", "8-FH-PMA", "8-FH", "8-PC-PMA", "8-PC", "8-UB-PMA", "8-UB", "9-AC-PMA", "9-AC", "9-FH-PMA", "9-FH", "9-PC-PMA", "9-PC", "10-Abd-PMA", "10-Abd", "10-FA-PMA", "10-FA", "10-FH-PMA", "10-FH", "10-UB-PMA",  "10-UB",  "11-AC-PMA",  "11-AC", "11-FA-PMA", "11-FA",  "11-FH-PMA", "11-FH", "11-PC-PMA",  "11-PC",  "11-UB-PMA",  "11-UB")

```

## plotting themes and functions

```{r}

theme_boxp <- function(){
  
  font <- "Times New Roman"
  
  theme_minimal() %+replace%
    
    theme(axis.title.y = element_text(angle = 90, size = 10), 
          axis.title.x = element_blank(),   
          axis.text.x = element_text(angle = 45, size = 10, color = "black"), 
          axis.text.y = element_text(size = 10, color = "black"), 
          
          panel.background = element_rect(fill = "white"),  
          axis.line.x = element_line(color = "black"), 
          axis.line.y = element_line(color = "black"), 
          
          legend.text = element_text(size = 10, lineheight = 0.6), 
          legend.spacing.y = unit(0.6, "mm"), 
          legend.key.height = unit(2, "mm"), 
          legend.key=element_rect(fill="white", color="black", linewidth = 0), 
          legend.title = element_blank(), 
          strip.background = element_rect(fill = NA, linetype = "solid", size = 1, colour = "black"), 
          strip.text = element_text(size = 10, margin = margin(1,1,1,1)), 
          panel.border = element_rect(fill = NA, size = 1),
          panel.spacing.y = unit(1,"line"),
    )
}

theme_pca <- function(){
  
  font <- "Times New Roman"
  
  theme_boxp() %+replace%
    
    theme(legend.position = "none", 
          axis.title.y = element_text(angle = 90, size = 12, color = "black"),
          axis.title.x = element_text(size = 12, color = "black"),
          axis.text.x = element_text(size = 10), 
          axis.text.y = element_text(size = 10),
          )
}

lm_eqn <- function(df){
    m <- lm(y ~ x, df);
    eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
         list(a = format(unname(coef(m)[1]), digits = 2),
              b = format(unname(coef(m)[2]), digits = 2),
             r2 = format(summary(m)$r.squared, digits = 3)))
    as.character(as.expression(eq));
}

```

## Import dataframe

```{r}

df <- read.delim2("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/skin_SMGC_Zebra60perc_formatted_genomelengthadjusted.tsv", check.names = FALSE)

df [, -1] <- sapply (df[, -1], as.numeric)

df <- as.data.frame(colSums(df[, -1]))

colnames(df) <- "Alignment_to_SMGC"

df$`sample-id` <- rownames(df)

df$Alignment_to_SMGC <- df$Alignment_to_SMGC

colnames(df)[2] <- "Sample"

```

## Metadata file

```{r}

metadata <- read.delim2("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/metadata.csv", check.names = FALSE, sep = ",")[-1, ]

colnames(metadata)[1] <- "Sample"

cells <- metadata[, c(1, 3, 6, 7, 25)]

cells[, c(-1:-3)] <- sapply(cells[, c(-1:-3)], as.numeric)

colnames(cells)[5] <- "total_cells"

```

## Correlation

```{r}

plot <- merge(df, cells, by = "Sample")

#plot <- plot[ ,colnames(plot) %in% c("Sample", "total_cells", "Alignment_to_SMGC")]

plot <- plot[ complete.cases(plot), ]
#plot[, -1] <- sapply(plot[, -1], as.numeric)


#1. Correlation between PMA sequencing reads and PMA treated flow cytometry
pma <- data.frame( 
          x = log(plot$Alignment_to_SMGC[ str_detect( plot$Sample, "-PMA")]),
          y = log(plot$total_cells[ str_detect( plot$Sample, "-PMA")]),
          BS = plot$BodySite_ID[ str_detect( plot$Sample, "-PMA")],
          ST = plot$Skin_Type[ str_detect( plot$Sample, "-PMA")]
  
)

#tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/Correlation_FCvsGenomLenAdj_PMA.tiff", units="in", width=4, height=4, res=300)
ggscatter(pma, x = "x", 
                  y = "y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "spearman",
          xlab = "log(total reads genome-length adjusted)", 
          ylab = "log(cell numbers)")
#dev.off()

#3. Correlation between untreated sequencing reads and untreated flow cytometry
pma <- data.frame( 
          x = log(plot$Alignment_to_SMGC[ !str_detect( plot$Sample, "-PMA")]),
          y = log(plot$total_cells[ !str_detect( plot$Sample, "-PMA")])
  
)

#tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/Corelation_FCvsGenomLenAdj_Raw.tiff", units="in", width=4, height=4, res=300)
ggscatter(pma, x = "x", 
                  y = "y", add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "spearman",
          xlab = "log(total reads genome-length adjusted)", 
          ylab = "log(cell numbers)")
#dev.off()


```

## Generatinglignment metrics table

```{r}

df <- read.delim2("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/alignment_stats_all.tsv.csv", check.names = FALSE)

df[, c(-1:-5)] <- sapply(df[, c(-1:-5)], as.numeric)

df$BodySite <- ordered(df$BodySite, levels = c( "Abd", "FA", "AC", "PC", "FH", "UB"))

df$SkinType <- ordered(df$SkinType, levels = c( "Sebaceous", "Moist", "Dry"))

df <- df[order(df$SkinType), ]

```

##plotting for mapped reads

```{r}


ggplot(df, aes( x = BodySite, y = df$`SwabBlank_Mapped%`, fill = SkinType)) +
        geom_boxplot() +
  geom_point(aes(color = Treatment), pch = 19) +
        theme_boxp() +
  theme(legend.position = "NULL") +
   scale_fill_manual(values = rev(c("indianred", "lightskyblue3", "darkgoldenrod"))) +
  scale_color_manual(values = c("Raw" = "tomato4", "PMA" = "moccasin")) +
        theme(legend.position = "NULL", panel.grid = element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11)) +
        labs(y = "% mapps reads", title = "Mapped to Swab_and_Kit")
       

tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/MappedPercent_toHuman.tiff", units="in", width=8, height=5, res=300)
ggplot(df, aes( x = BodySite, y = df$`Host_Mapped%`*100, fill = SkinType)) +
       geom_boxplot() +
  geom_point(aes(color = Treatment), pch = 19) +
        theme_boxp() +
  theme(legend.position = "NULL") +
   scale_fill_manual(values = rev(c("indianred", "lightskyblue3", "darkgoldenrod"))) +
  scale_color_manual(values = c("Raw" = "tomato4", "PMA" = "moccasin")) +
        theme(legend.position = "NULL", panel.grid = element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11))  +
        labs(y = "% mapps reads", title = "Mapped to Host ref genome (GRCh39.p14)")
dev.off()

tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/MappedPercent_toSMGC.tiff", units="in", width=8, height=5, res=300)
ggplot(df, aes( x = BodySite, y = df$`SMGC_Mapped%`*100, fill = SkinType)) +
  geom_boxplot() +
  geom_point(aes(color = Treatment), pch = 19) +
        theme_boxp() +
  theme(legend.position = "NULL") +
   scale_fill_manual(values = rev(c("indianred", "lightskyblue3", "darkgoldenrod"))) +
  scale_color_manual(values = c("Raw" = "tomato4", "PMA" = "moccasin")) +
        theme(legend.position = "NULL", panel.grid = element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11))  +
        labs(y = "% mapps reads", title = "Mapped to SMGC") +
  stat_compare_means()
dev.off()



```

## Reads passing each filtering step and mapping 

```{r}

df_SMGC <- aggregate(df[, str_detect(colnames(df), "Reads")], by = list(df$BodySite, df$Treatment), mean)

df_SMGC$Group.1 <- ordered(df_SMGC$Group.1, levels = c( "Abd", "FA", "AC", "PC", "FH", "UB"))

tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/Reads_passing_filters_mappingSMGC.tiff", units="in", width=8, height=5, res=300)
ggplot(melt(df_SMGC[, !str_detect(colnames(df_SMGC), "WoL")]), aes(x = `Group.1`, y = value, fill = variable)) +
  geom_bar(position = "dodge", stat = "identity", color = "black") + 
  scale_fill_brewer(palette = "Set3") + 
  facet_wrap(~Group.2) +
  labs(
    title = "Average reads retained following filtering steps",
    x = "",
    y = "",
    fill = ""
  ) +
  theme_minimal(base_size = 12) + 
  theme_boxp() +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
    legend.position = "top",  # Place the legend at the top
    axis.text.x = element_text(angle = 45, hjust = 1)  # Rotate x-axis labels for better readability
  )
dev.off()

df <- melt(df_SMGC)

df$group <- df$variable

df$variable <- paste(df$variable, df$Group.2, sep = "-")

tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/Reads_passing_filters_mappingSMGC_vs_WoL.tiff", units="in", width=9, height=5, res=300)
ggplot(df, aes(x = `Group.1`, y = value, fill = group)) +
  geom_bar(position = "dodge", stat = "identity", color = "black") + 
  scale_fill_brewer(palette = "Set3") + 
  facet_wrap(~Group.2) +
  labs(
    title = "Average reads retained following filtering steps",
    x = "",
    y = "",
    fill = ""
  ) +
  theme_minimal(base_size = 12) + 
  theme_boxp() +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
    legend.position = "top",  # Place the legend at the top
    axis.text.x = element_text(angle = 45, hjust = 1)  # Rotate x-axis labels for better readability
  )
dev.off()

compare_means(value~variable, df,  method = "wilcox", p.adjust.method = "fdr")

tiff("/media/deepan/Deepan/LiveDead/igm-storage2.ucsd.edu/220428_A00953_0539_AHMNLKDSX3/202410_revision/swab_blank_removed/TIFFs/Mapping_SMGC_vs_WoL.tiff", units="in", width=9, height=5, res=300)
ggplot(df[df$group %in% c("SMGC_Mapped_Reads", "WoL_Mapped_Reads"),], aes(x = `Group.1`, y = value, fill = group)) +
  geom_bar(position = "dodge", stat = "identity", color = "black") + 
  scale_fill_brewer(palette = "Set3") + 
  facet_wrap(~Group.2) +
  labs(
    title = "Average reads retained following filtering steps",
    x = "",
    y = "",
    fill = ""
  ) +
  theme_minimal(base_size = 12) + 
  theme_boxp() +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 14),
    legend.position = "top",  # Place the legend at the top
    axis.text.x = element_text(angle = 45, hjust = 1)  # Rotate x-axis labels for better readability
  )
dev.off()

```


